planner/core/rule_join_reorder.go

```diff
@@ -165,7 +170,35 @@ func (s *joinReOrderSolver) optimizeRecursive(ctx sessionctx.Context, p LogicalP
 				break
 			}
 		}
-		if len(curJoinGroup) > ctx.GetSessionVars().TiDBOptJoinReorderThreshold || !isSupportDP {
+
+		baseGroupSolver := &baseSingleGroupJoinOrderSolver{
+			ctx:        ctx,
+			otherConds: otherConds,
+			eqEdges:    eqEdges,
+			joinTypes:  joinTypes,
+		}
+
+		joinGroupNum := len(curJoinGroup)
+		useGreedy := joinGroupNum > ctx.GetSessionVars().TiDBOptJoinReorderThreshold || !isSupportDP
+
+		if hintInfo != nil && hintInfo.leadingJoinOrder != nil {
+			if hasOuterJoin {
+				ctx.GetSessionVars().StmtCtx.AppendWarning(ErrInternal.GenWithStack("leading hint is inapplicable when we have outer join"))
```

Comment:
Please add some cases for it.

Reply:
https://github.com/pingcap/tidb/pull/34570/files#diff-10b0e4de9f7148711328e0180170bcba713ad40b4e9234b16556c626b1a6b2eaR80 Here is the test cases for outer join